Sources/TuistCore/Models/TuistProject.swift

```diff
@@ -0,0 +1,46 @@
+import XcodeGraph
+
+public enum TuistProject: Equatable, Hashable {
+    /// Creates a configuration for a Tuist project.
+    ///
+    /// - Parameters:
+    ///   - compatibleXcodeVersions: List of Xcode versions the project is compatible with.
+    ///   - swiftVersion: The version of Swift that will be used by Tuist.
+    ///   - plugins: A list of plugins to extend Tuist.
+    ///   - generationOptions: List of options to use when generating the project.
+    ///   - installOptions: List of options to use when running `tuist install`.
+    case generated(
```

Comment:
The public version of this is `.tuist`, which doesn't reflect the recent decision to use "generated projects" instead of "tuist projects". Since changing the public interface would be a breaking change, I decided to do it only internally.